src/Bloc/BlAnnulusSectorGeometry.class.st

```diff
@@ -83,44 +128,63 @@ BlAnnulusSectorGeometry >> matchExtent: aPoint [
 { #category : #accessing }
 BlAnnulusSectorGeometry >> outerRadius [
 	"An outer radius as a fraction of the max radius of the annulus [0..1]"
-	<return: #Number>
 
 	^ outerRadius
 ]
 
 { #category : #accessing }
 BlAnnulusSectorGeometry >> outerRadius: aNumber [
 	"Set outer radius as a fraction of the max radius of the annulus [0..1]"
+
 	outerRadius := aNumber min: 1.0 max: 0.0.
 	self releaseCache
 ]
 
-{ #category : #'private - geometry' }
-BlAnnulusSectorGeometry >> pointFromAngle: anAngleInRadians andRadius: aRadius around: aCenter [
-	"Convert from spherical coordinates to cartesian coordinates"
-	<return: #Point>
-	| aX aY |
-	
-	aX := aCenter x + (aRadius * anAngleInRadians cos).
-	aY := aCenter y + (aRadius * anAngleInRadians sin).
-
-	^ aX @ aY
-]
-
 { #category : #'private - geometry' }
 BlAnnulusSectorGeometry >> radius [
-	<return: #Number>
 	
 	^ self extent min / 2.0
 ]
 
 { #category : #accessing }
 BlAnnulusSectorGeometry >> startAngle [
-	^ startAngle
+
+	^ startAngleInRadians radiansToDegrees
 ]
 
 { #category : #accessing }
 BlAnnulusSectorGeometry >> startAngle: anAngleInDegrees [
-	startAngle := anAngleInDegrees.
+
+	self startAngleInRadians: anAngleInDegrees degreesToRadians
+]
+
+{ #category : #accessing }
+BlAnnulusSectorGeometry >> startAngle: anAngle endAngleDelta: anotherAngle [
+	"Establish the start and end angles of this sector, by passing the starting angle expressed as degrees, and a delta. A positive delta is a clockwise sector, while negative counter-clockwise."
+
+	self
+		startAngle: anAngle;
+		endAngle: anAngle + anotherAngle
+]
+
+{ #category : #accessing }
+BlAnnulusSectorGeometry >> startAngleInRadians [
+
+	^ startAngleInRadians
+]
+
+{ #category : #accessing }
+BlAnnulusSectorGeometry >> startAngleInRadians: anAngleInRadians [
+
+	startAngleInRadians := anAngleInRadians.
 	self releaseCache
 ]
+
+{ #category : #accessing }
+BlAnnulusSectorGeometry >> startAngleInRadians: anAngleInRadians endAngleDelta: anotherAngleInRadians [
+	"Establish the start and end angles of this sector, by passing the starting angle expressed as radians and a delta. A positive delta is a clockwise sector, while negative counter-clockwise."
+
+	self
+		startAngleInRadians: anAngleInRadians;
+		endAngleInRadians: anAngleInRadians + anotherAngleInRadians
+]
```

Comment:
This PR was ready to merge, but I wanted to think a bit more before introducing new API in this geometry. I think it's a bit annoying the multiplication of variants in the API seen in this fragment of the file. We may add a BlAngle to specify angles either on degrees or radians, using a uniform API. 

For example, in CSS you can specify in several angle units:
deg: Represents an angle in degrees. One full circle is 360deg. Examples: 0deg, 90deg.
grad: Represents an angle in gradians. One full circle is 400grad. Examples: 0grad, 38.8grad.
rad: Represents an angle in radians. One full circle is 2π radians which approximate to 6.2832rad. 1rad is 180/π degrees. Examples: 0rad, 1.07rad.
turn: Represents an angle in a number of turns. One full circle is 1turn. Examples: 0turn, 1.2turn.

We already have reifications such as BlLogicalDisplayPosition and BlLogicalDisplayPosition, as well as BlLogicalDisplaySize and BlPhysicalDisplaySize. Angle seems good. There is already GAngle, in the Geometry package, which is quite complete but we don't depend on this project. A tiny BlAngle can be enough.

Opinions?

Reply:
So instead of passing around scalars, e.g. 90deg, or Pi/2, or 100grad, or 0.25 turns, we would pass around a first-class object: BlAngle.  If I understand you correctly, that sounds like a very good idea.   I particularly like representing the values as [0.0,1.0) for 1 turn.  That's great!